ext/compressors/zstd/zstd_compress.c

```diff
@@ -44,13 +44,36 @@
 #define inline __inline
 #endif
 
+/* Default context pool size. */
+#define CONTEXT_POOL_SIZE 50
+
+struct ZSTD_Context;
+typedef struct ZSTD_Context ZSTD_CONTEXT;
+struct ZSTD_Context {
+    void *ctx; /* Either a compression context or a decompression context. */
+    ZSTD_CONTEXT *next;
+};
+
+struct ZSTD_Context_Pool;
+typedef struct ZSTD_Context_Pool ZSTD_CONTEXT_POOL;
+struct ZSTD_Context_Pool {
+    int count;       /* Pool size */
+    void *list_lock; /* Spinlock */
+    ZSTD_CONTEXT *free_ctx_list;
+};
+
+typedef enum { CONTEXT_TYPE_COMPRESS, CONTEXT_TYPE_DCOMPRESS } CONTEXT_TYPE;
```

Comment:
Could you change `CONTEXT_TYPE_DCOMPRESS` to `CONTEXT_TYPE_DECOMPRESS`? Or does your chosen name match another usage somewhere?